platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3865,13 +3890,35 @@
                 callback);
         };
 
-        $scope.setAvailableFieldsForDisplay = function() {
+        $scope.subclassCurrent = null;
+
+        $scope.setAvailableFieldsForDisplay = function(scope) {
             var i;
-            $scope.availableFieldsForDisplay = [];
-            for (i = 0; i < $scope.allEntityFields.length; i += 1) {
-                if (!$scope.allEntityFields[i].nonDisplayable) {
-                    $scope.availableFieldsForDisplay.push($scope.allEntityFields[i]);
+            var availableFieldsForDisplay = [];
+            for (i = 0; i < scope.allEntityFields.length; i += 1) {
+                if (!scope.allEntityFields[i].nonDisplayable) {
+                    availableFieldsForDisplay.push(scope.allEntityFields[i]);
+                }
+            }
+            var subclassnameCurrent = scope.$parent.$parent.$parent.subclassCurrent;
```

Comment:
Thanks for findCurrentScope().  I was unable to tell whether or not the existing code did a close enough action to work for my reuse.